example/lib/main.dart

```diff
@@ -12,32 +10,15 @@ class MyApp extends StatefulWidget {
 }
 
 class _MyAppState extends State<MyApp> {
-  String _platformVersion = 'Unknown';
 
-  @override
-  void initState() {
-    super.initState();
-    initPlatformState();
+  Future<void> trackAction(String action, Map<String, String> data, {@required BuildContext context}) async {
+    final success = await AdobeExperiencePlatform.trackAction(action, data);
+    print("Action tracking result : ${success ? "success" : "failure"}");
   }
 
-  // Platform messages are asynchronous, so we initialize in an async method.
-  Future<void> initPlatformState() async {
-    String platformVersion;
-    // Platform messages may fail, so we use a try/catch PlatformException.
-    try {
-      platformVersion = await FlutterAdobeExperiencePlatformPlugin.platformVersion;
-    } on PlatformException {
-      platformVersion = 'Failed to get platform version.';
-    }
-
-    // If the widget was removed from the tree while the asynchronous platform
-    // message was in flight, we want to discard the reply rather than calling
-    // setState to update our non-existent appearance.
-    if (!mounted) return;
-
-    setState(() {
-      _platformVersion = platformVersion;
-    });
+  Future<void> trackState(String state, Map<String, String> data, {@required BuildContext context}) async {
+    final success = await AdobeExperiencePlatform.trackState(state, data);
```

Comment:
`AdobeExperiencePlatform` is too broad right?
We wanted to split this up in multiple extensions.
Shouldn't this be part of the `AdobeAnalytics` extension?

Reply:
Never mind just saw the next PR